### PR TITLE
Update API functions

### PR DESCRIPTION
**Pull request by LaszloLango**
* Rename 'jerry_api_' prefix to 'jerry_'
 * Fix minor style issues
 * Group the API functions and add comment to each group
 * Move engine behaviour related funtions to 'jerry.h'

JerryScript-DCO-1.0-Signed-off-by: László Langó llango.u-szeged@partner.samsung.com

### CODE REVIEW

**Comment by LaszloLango** [comment]
If we accept this PR, then I'll add an additional commit to the PR that updates the names in the related documentations.

File: jerry-core/jerry.c [hunk 1]
```diff
@@ -125,251 +125,251 @@ jerry_make_api_unavailable (void)
...
  *
  * Note:
  *      if the output value contains string / object, it should be freed
- *      with jerry_api_release_string / jerry_api_release_object,
+ *      with jerry_release_string / jerry_release_object,
  *      just when it becomes unnecessary.
  */
 static void
-jerry_api_convert_ecma_value_to_api_value (jerry_api_value_t *out_value_p, /**< [out] api value */
+jerry_convert_ecma_value_to_api_value (jerry_value_t *out_value_p, /**< [out] api value */
```
Comment (zherczeg) [inline jerry-core/jerry.c]: Alignment issue

**Comment by zherczeg** [comment]
Good. LGTM

**Comment by dbatyai** [comment]
LGTM

**Pull request merged**